arc_cache.go

```diff
@@ -47,6 +50,9 @@ func (b *arccache) DeleteBlock(k cid.Cid) error {
 		return nil
 	}
 
+	b.lks[k.Bytes()[len(k.Bytes())-1]].Lock()
```

Comment:
1. FYI, `k.Bytes()` is kind of inefficient (converts to a byte slice which will likely allocate). It's better to just use `k.KeyString()[...]`.
2. Assuming the CID is "defined" (non-zero length) is probably too risky. We should have a quick check above.

Reply:
We don't _currently_ bother with that check for 2 because it didn't really matter. We need it now because we can panic otherwise.

Reply:
We don't _currently_ bother with that check for 2 because it didn't really matter. We need it now because we can panic otherwise.